### PR TITLE
Add annotation guidelines

### PR DESCRIPTION
**Pull request by Garanas**

### CODE REVIEW

**Comment by hahn-kev** [comment]
Looks good, I would provide some examples for how we annotate classes. Both the class definition, and how we also need to annotate all the functions on a class.

**Comment by Garanas** [comment]
Took into account your feedback @hahn-kev , is that what you intended?

File: annotation.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+--- | TransportReturn    | Location for transports to return to (they will attack with the land units if this isn't set)
+--- | TransportChain     | (REQUIRED or TransportRoute)
+--- | AttackPoints       | (REQUIRED or AttackChain or PatrolChain) The platoon attacks the highest threat first
+--- | AttackChain        | (REQUIRED or AttackPoints or PatrolChain)
+--- | PatrolChain        | (REQUIRED or AttackChain or AttackPoints)
+---@param platoon Platoon
+function LandAssaultWithTransports(platoon)
+```
+
+Note that the latter is from `scenarioplatoonai.lua` and that file contains a series of platoon behaviors. Because of its setup, the parameters of the function are send via `platoon.PlatoonData` instead of as separate parameters. We can't change this - but via a markdown table we can still help the user to understand what parameters are there.
```
Comment (hahn-kev) [inline annotation.md:58]: `parameters of the function are send via`
'send' should be 'sent' here

**Comment by hahn-kev** [comment]
> Took into account your feedback @hahn-kev , is that what you intended?

yeah that looks good. I noticed one correction of English when I re-read it. Otherwise it looks good to me.

**Comment by Garanas** [comment]
The russian translation is coming in via Resistance, I'll add it to this PR once I'm back

**Comment by Garanas** [comment]
Translation via Resistance, by WhiteOwl:

```md
87: Для установки вам необходимо скачать файл “.vsix” и установить его в VS Code. Следуйте инструкции на картинке. Вы можете выбрать расширение загружаемого файла
89: Каждый раз, когда появляется новая версия расширения, вам необходимо скачать и установить его снова
91: После установки вы можете приступить к FAF repo?  Для папок модов или карт процесс установки немного другой. (TODO, напишите инструкцию для карт/модов)
93: Использовать отладчик в игре
95: Игровой отладчик снова работает
97: Для использования отладчика Lua вам необходимо использовать:
99: - Комбинация клавиш по умолчанию (Иногда работает только после использования меню привязок)
100: В игровой консоли 
104: Устанавливает точку остановки на строке
105: Возобновить выполнение
106: (не могу найти способ сделать шаг назад)
112: Если у вас есть точка остановки в часто используемых включениях, это может в целом сильно замедлить игру, даже в случае несрабатываея стоп-точки. В этом случае лучше отключить или удалить все точки остановки, а затем поставить их тогда, когда они понадобятся 
113: Из-за этого иногда кажется, что игра виснет. Вы можете подождать или закрыть игру и очистить все точки остановки после этого
120: Подключение отладчика
123: Этот шаг не является обязательным и требуется только в случае, если хотите проверить исключение/сбой на рабочем столе
125: Благодаря KionX у нас есть отладчик. Когда появляется исключение, отладчик может отследить его до строки кода Lua. Исполняемый скомпилированный файл доступен в каждом релизе. Сохраните исполняемый файл в папке “bin”. Это та самая папка которая была обозначена ранее. Нам нужно адаптировать файлы bat/bash для использования отладчика. В качестве примера мы меняем файл “bat” на:
135: Аргументы передаются отладчиком. Изменение в скрипте bash аналогично. Когда игра крашится, отладчик будет пытаться сообщить в логе что случилось.
137: Воспроизвести реплей
140: Жёсткий сбой может появиться только в реплее. Для выяснения необходимо использовать отладчик. Можете запустить реплей используя отладчик. Для этого необходимо два шага: сравнить версию игры реплея и получить сам реплей?
142: Версия игры зависит от типа игры. Извлечь repository в нужную ветку:
147: В отличие от файлов “.fafreplay” которые могжно получить и просмотреть с помощью клиента faf, базовая игра которая запускается для теста и отладки распознает только расширение “.scfareplay” для реплеев. Для конвертации `.fafreplay` в`.scfareplay` можно запустить воспроизведение с помощью клиета faf и сразу же закрыть его. Клиент создаст временную версию повтора в формате `.scfareplay` в папке кэша клиента:
150: Скопируйте этот реплей в папку реплеев игры:
153: Обратите внимание что последний путь незакончен: вам необходимо заменить `%USER_NAME%` на имя профиля вашей системы, а `%PROFILE_NAME%` на имя профиля который вы используете в игре. Вы можете запустить игру с помощью bat файла как это было описано.
155: Несколько экземпляров
158: Вы можете запустить несколько процессов одновременно когда отладка будет включена. Вы можете сделать это добавлением этой строки в файл настроек:
161: Файл настроек можно найти здесь:
167: Все файлы в каталоге `schook` перехватывают файлы в папке `lua`. Мод применяет аналогичную стратегию перехвата. Перехват означает объединение файлов. Один файл добавляется к следующему. Это позволяет добавлять код в файл. В качестве примера предположим, что у нас есть этот базовый файл игры:
170: И мы пытаемся подключить его к обоим:
174: Произойдёт следующее:
177: Где оператор присоединения следует интерпретировать как добавление одного текстового (исходного) файла к другому текстовому (исходному) файлу. это расширит файл модуля. Это присоединение будет выполнено когда файл импортируется в первый раз. Каталог который используется для подключения можно настроить в файлах инициализации.
182: Полезные горячие клавиши и ресурсы которые могут быть использованы в игре:
184: Используются для вызова окна спавна юнитов. Хорошо подходит для свободного управления или для управления под руководством AI.
185: Отрыть окно журнала если вы закрыли его или оно было закрыто по какой-то причине
186: Открыть консоль
187: В консоле использовать: `ren_ShowNetworkStats` для вызова sim rate / package drops в правом верхнем углу
188: В консоле использовать: `ShowStats` для вызова engine stats в левой части экрана
```

If I am correct the line numbers reflect the line numbers in `Files changed`, @MrRowey could I encourage you to add this to the PR?

**Comment by MrRowey** [comment]
@Garanas what PR ?

**Comment by Garanas** [comment]
To this pull request, match the update of the russian section (using the translations) with the english section

**Comment by MrRowey** [comment]
> To this pull request, match the update of the russian section (using the translations) with the english section

i'm not sure if i'm complete overlooking a file but the number in the translation don't match with any files changed in this PR

**Comment by Garanas** [comment]
You're correct, it matches https://github.com/FAForever/fa/pull/3984/files

I send Resistance the wrong PR 😄 , can you add that in then? Via a new PR?

**Comment by Garanas** [comment]
@Hdt80bro in the future we can adjust these guidelines, but for now I'm merging this as otherwise it keeps lingering on

**Pull request merged**